### DISCUSSION TITLE
How do share data between page shifts?

### DISCUSSION THREAD

**Discussion by livevsonline** [TanStack/router#563] [Q&A]
Hi i"m trying to share data between pages, basically i want to check if the "systeminfo" is already set, and if it is i don"t want to load a new one?

Im using:

"@tanstack/react-router": "beta",
"@tanstack/react-loaders": "beta",

Im a bit unsure on how to do  this, as it seem like the context is wiped on every new route visit?

2 Routes/Loaders

```
const salesRunningRoute = new Route({
	id: 'salesRunning',
	getParentRoute: () => rootRoute,
	path: '/app/sales/running',
	async onLoad({context}) {
		return await context.loaderClient.getLoader({key: salesRunningLoader.key})
			.load({
				variables: () => context,
			})
			.then((e) => {
				if (context.systemInfo === undefined) {
					console.log('WE NEED TO SET SYSTEM INFO 1');
				}
			});
	},
	component: (e) => {
		return <SalesRunning/>;
	},
	// component: lazy(() => import("@/pages/sales/running/SalesRunning")),
});

export const salesRunningLoader = new Loader({
	key: 'salesRunningLoader',
	loader: async (params: () => RouteContextType) => {
		const oContext = params();
		return {
			systemInfo: oContext.systemInfo=== undefined ? 'system info 1: ' : oContext.systemInfo
		};
	},
});

const runningAndShippedRoute = new Route({
	id: 'purchaseRunningShipped',
	path: '/app/purchase/running/shipped',
	getParentRoute: () => rootRoute,
	async onLoad({context, params}) {
		console.log('ROUTE RUNNING AND SHIPPED CONTEXT: ', context);

		return await context.loaderClient.getLoader({key: runningAndShippedLoader.key})
			.load({
				variables: () => context,
			})
			.then((e) => {
				if (context.systemInfo === undefined) {
					console.log('WE NEED TO SET SYSTEMINFO 2');
				}
			});
	},
	component: (e) => {
		return <RunningAndShipped/>;
	},
});

export const runningAndShippedLoader = new Loader({
	key: 'purchaseRunningShippedLoader',
	loader: async (params: () => RouteContextType) => {
		const oContext = params();
		return {
			systemInfo: oContext.systemInfo=== undefined ? 'system info 2: ' : oContext.systemInfo
		};
	},
});
```

Other Route Config

```
export const loaderClient = new LoaderClient({
	getLoaders: () => {
		return [
			runningAndShippedLoader,
			salesRunningLoader,
		]
	},
});

type RouteContextType = {
	loaderClient: typeof loaderClient,
	systemInfo?: ResponseSystemInfoData,
}

const rootRoute = RootRoute.withRouterContext<RouteContextType>()({
	errorComponent: () => <ErrorLoadingPage/>
});


const routeTree = rootRoute.addChildren([
	salesRunningRoute,
	runningAndShippedRoute,
	testRoute
]);

export const router = new ReactRouter({
	routeTree,
	defaultPreload: 'intent',
	context: {
		loaderClient,
		systemInfo: undefined,
	},
});

```

Currently im passing contexts to the onLoad->contextLoadcClient->load function, which i think is wrong, but how do i do it then?

Basically all i want is  to set a flag if "X" has already been loaded/not undefined, then load it otherwise use the defined value?

**Comment by livevsonline** [comment]
Would also be nice if the documentation would reflect a  specific package tag/version, as im getting soooo many errors trying to follow the documentation
